File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/seat/SeatUnselectDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.seat;

import at.ac.tuwien.sepm.groupphase.backend.entity.enums.Type;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import javax.validation.constraints.Min;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SeatUnselectDto {

    @NonNull
    private Long ticketId;

    @NonNull
    private Type type;

    @Nullable
    @Min(0)
    private Integer row;

    @Nullable
    @Min(0)
    private Integer col;

    @Nullable
    private Long areaId;

}

File: frontend/src/app/dtos/seat/seat-unselect-dto.ts
import { SeatType } from "../seat-type";

export class SeatUnselectDto {

  private _row: number;
  private _col: number;
  private _areaId: number;

  constructor(private _ticketId: number, private _type: SeatType) {
  }

  set row(value: number) {
    this._row = value;
  }

  set col(value: number) {
    this._col = value;
  }

  set areaId(value: number) {
    this._areaId = value;
  }

  set ticketId(value: number) {
    this._ticketId = value;
  }

  set type(value: SeatType) {
    this._type = value;
  }

  get row(): number {
    return this._row;
  }

  get col(): number {
    return this._col;
  }

  get areaId(): number {
    return this._areaId;
  }

  get ticketId(): number {
    return this._ticketId;
  }

  get type(): SeatType {
    return this._type;
  }
}
